core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4

```diff
@@ -3190,7 +3190,7 @@ jsonLiteral: LCB kvPair (COMMA kvPair)* RCB | LCB RCB
 kvPair: key = string COLON literal
     ;
 
-arrayLiteral: LSB literal (COMMA literal)* RSB | LSB RSB
+arrayLiteral: LSB expr (COMMA expr)* RSB | LSB RSB
```

Comment:
Instead doing this can we make Literal Schema Aware?

Reply:
Rough idea

```
literal
    : DATE string
    | TIMESTAMP string
    | string
    | sign? DECIMAL
    | sign? (REAL | FLOAT)
    | trueFalse
    | jsonLiteral
    | arrayLiteral
    | NULL_
    | string :: cast
    | PARAM // A question mark can be used as a placeholder for a prepared statement that will use binding.
    ;
```

Reply:
@sundarshankar89 separate PR, also - `ARRAY(CURENT_USER(), 'foo')` is a valid literal of `ARRAY<STRING>`.